src/AssetsController.ts

```diff
@@ -419,13 +423,26 @@ export class AssetsController extends BaseController<AssetsConfig, AssetsState>
 		address = toChecksumAddress(address);
 		const { allCollectibles, collectibles } = this.state;
 		const { networkType, selectedAddress } = this.config;
-		const newCollectibles = collectibles.filter(
-			(collectible) => !(collectible.address === address && collectible.tokenId === tokenId)
+		const ignoredCollectible = collectibles.find(
+			(collectible) => collectible.address === address && collectible.tokenId === tokenId
 		);
+
+		const ignoredCollectibles = [...this.state.ignoredCollectibles];
+		/* istanbul ignore else */
+		if (ignoredCollectible) {
+			ignoredCollectibles.push(ignoredCollectible);
+		}
+
+		const newCollectibles = collectibles.filter((collectible) => {
+			return !(collectible.address === address && collectible.tokenId === tokenId);
+		});
+
 		const addressCollectibles = allCollectibles[selectedAddress];
 		const newAddressCollectibles = { ...addressCollectibles, ...{ [networkType]: newCollectibles } };
 		const newAllCollectibles = { ...allCollectibles, ...{ [selectedAddress]: newAddressCollectibles } };
-		this.update({ allCollectibles: newAllCollectibles, collectibles: newCollectibles });
+
+		this.update({ ignoredCollectibles });
```

Comment:
didn't you have issues from the app doing this? we used to have a bug with states being updated like this before where states weren't being updated. We solved it changing all `update`s to the format `this.update({ ignoredCollectibles: [...ignoredCollectibles]})`

Reply:
also we could add it to the `update` below to avoid two state updates